src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -771,6 +774,9 @@ public void deleteResultsAndSubmissionsOfParticipation(Long participationId, boo
         submissions.forEach(submission -> {
             if (submission instanceof ProgrammingSubmission) {
                 coverageReportRepository.deleteBySubmissionId(submission.getId());
+                // This clears the build log entries and deletes them from the corresponding table
+                ((ProgrammingSubmission) submission).setBuildLogEntries(List.of());
```

Comment:
how about `new ArrayList()` instead of `List.of()` ?

Reply:
`List.of()`/`Collections.emptyList()` is an unmodifiable singleton-instance, so no object allocation is necessary.

`new ArrayList<>()` creates a new initial array with some capacity.

Reply:
I'd prefer `Collections.emptyList()` here since it clrearly shows that the list is intentianally empty. With `List.of()`, another developer could just add some elements as parameters.

Reply:
I chose `Collections.emptyList()` now because it shows that the list should always be empty.